src/Components/StatusDropDown/StatusDropDown.tsx

```diff
@@ -0,0 +1,117 @@
+import React from "react";
+import { StyleSheet, Text, ViewStyle, View } from "react-native";
+import { TransparentButton } from "src/Components/Buttons";
+import getShadowStyle from "src/Themes/getShadowStyle";
+import { getTransparentColor } from "src/Themes/Colors";
+import { Point, TriangleDown, TriangleUp } from "src/Components/Icons";
+import withTheme, { ThemeProps, ThemeParamsType } from "src/Themes/withTheme";
+
+type Status = { key: string; value: React.ReactNode };
+
+interface OwnProps {
+  currentStatus: Status;
+  dropDownStatuses: Status[];
+  onStatusPress(statusKey: string): void;
+  color: string;
+  style?: ViewStyle;
+  statusIcon?: React.ReactNode;
+}
+
+type Style = ReturnType<typeof getStyles>;
+
+interface Props extends OwnProps, ThemeProps<Style> {}
+
+interface State {
+  isOpen: boolean;
+}
+
+export class PureStatusDropDown extends React.PureComponent<Props, State> {
+  public state = {
+    isOpen: false,
+  };
+  public render() {
+    const { styles } = this.props;
+    return (
+      <View>
+        {this.renderDropDownMenu()}
+        <TransparentButton
+          onPress={this.toggleDropDown}
+          style={[styles.container, { backgroundColor: getTransparentColor(this.props.color) }, this.props.style]}>
+          <View style={styles.statusContainer}>
+            <Point color={this.props.color} width={6} height={6} />
+            <View style={styles.statusIconContainer}>{this.props.statusIcon}</View>
+            <Text style={[styles.statusLabel, { color: this.props.color }]}>{this.props.currentStatus.value}</Text>
+          </View>
+          {this.state.isOpen ? (
+            <TriangleUp color={this.props.color} width={12} height={12} />
+          ) : (
+            <TriangleDown color={this.props.color} width={12} height={12} />
+          )}
+        </TransparentButton>
+      </View>
+    );
+  }
+
+  private renderDropDownMenu = () => {
+    if (!this.state.isOpen) return null;
+    const { styles, colors } = this.props;
+    return (
+      <View style={styles.dropDownMenuContainer}>
+        {this.props.dropDownStatuses.map((status) => {
+          const isCurrentStatus = status.key === this.props.currentStatus.key;
+          return (
+            <TransparentButton
```

Comment:
It will be better to create a separate component here to make the code a little bit nicer by isolating the internal components' logic to a separate function.

`isCurrentStatus` can be passed from outside one time (!!), so you will no need to create a separate variable to store this value. And you will be able to use one-line-component return style for a map function: `.map((status) => (<Foo />))` instead of `.map((status) => { return (<Foo />) })`